apps/asap-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts

```diff
@@ -0,0 +1,170 @@
+import { APIGatewayProxyResult } from 'aws-lambda';
+import Boom from '@hapi/boom';
+import { apiGatewayEvent } from '../../../helpers/events';
+import {
+  algoliaSearchApiKey,
+  auth0SharedSecret as secret,
+} from '../../../../src/config';
+import { identity } from '../../../helpers/squidex';
+import { fetchUserByCodeHandlerFactory } from '../../../../src/handlers/webhooks/fetch-by-code/fetch-by-code';
+import { SearchClient } from 'algoliasearch';
+import { userControllerMock } from '../../../mocks/user-controller.mock';
+import { userResponse } from '../../../fixtures/users.fixtures';
+
+describe('Fetch-user-by-code handler', () => {
+  const algoliaClientMock = {
+    generateSecuredApiKey: jest.fn(),
+  } as unknown as jest.Mocked<SearchClient>;
+  const handler = fetchUserByCodeHandlerFactory(
+    userControllerMock,
+    algoliaClientMock,
+  );
+
+  describe('Validation', () => {
+    test("return 400 when code isn't present", async () => {
+      const result = (await handler(
+        apiGatewayEvent({
+          headers: {
+            Authorization: `Basic ${secret}`,
+          },
+        }),
+      )) as APIGatewayProxyResult;
+
+      expect(result.statusCode).toStrictEqual(400);
+    });
+
+    test('returns 401 when request is not authorized', async () => {
+      const result = (await handler(
+        apiGatewayEvent({
+          pathParameters: {
+            code: 'welcomeCode',
+          },
+        }),
+      )) as APIGatewayProxyResult;
+
+      expect(result.statusCode).toStrictEqual(401);
+    });
+
+    test('returns 401 when request is not using Basic Auth', async () => {
```

Comment:
If possible can we use digest or bearer auth here instead?

Reply:
we could, but probably not in this PR, as this PR does not address auth0 auth method. Feel free to raise as a Tech debt though